pymc3/distributions/timeseries.py

```diff
@@ -206,15 +213,46 @@ def logp(self, x):
         -------
         TensorVariable
         """
-        sigma = self.sigma
-        mu = self.mu
-        init = self.init
+        if x.ndim > 0:
+            x_im1 = x[:-1]
+            x_i = x[1:]
+            if self.sigma.ndim > 0:
+                sigma = self.sigma[:-1]
+            else:
+                sigma = self.sigma
+            if self.mu.ndim > 0:
+                mu = self.mu[:-1]
+            else:
+                mu = self.mu
 
-        x_im1 = x[:-1]
-        x_i = x[1:]
+            innov_like = Normal.dist(mu=x_im1 + mu, sigma=sigma).logp(x_i)
+            return self.init.logp(x[0]) + tt.sum(innov_like)
+        return self.init.logp(x)
+
+    def random(self, point=None, size=None):
+        """Draw random values from GaussianRandomWalk.
 
-        innov_like = Normal.dist(mu=x_im1 + mu, sigma=sigma).logp(x_i)
-        return init.logp(x[0]) + tt.sum(innov_like)
+        Parameters
+        ----------
+        point : dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size : int, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+        """
+        sigma, mu = distribution.draw_values([self.sigma, self.mu], point=point, size=size)
+        return distribution.generate_samples(self._random, sigma=sigma, mu=mu, size=size,
+                                             dist_shape=self.shape)
+
+    def _random(self, sigma, mu, size):
+        """Implement a Gaussian random walk as a cumulative sum of normals."""
+        rv = stats.norm(mu, sigma)
+        return rv.rvs(size).cumsum(axis=0)
```

Comment:
Careful with `axis=0`. If `mu` is an RV, its drawn value will have the `size` prepend, and that will shift the time series axis further to the right.